src/test/kotlin/com/edd/memegrid/web/RouterTest.kt

```diff
@@ -0,0 +1,285 @@
+package com.edd.memegrid.web
+
+import com.edd.memegrid.memes.Meme
+import com.edd.memegrid.memes.MemeManager
+import com.edd.memegrid.util.*
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.mockk.clearMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.restassured.RestAssured.given
+import io.restassured.http.ContentType
+import io.restassured.module.kotlin.extensions.Extract
+import io.restassured.module.kotlin.extensions.Given
+import io.restassured.module.kotlin.extensions.Then
+import io.restassured.module.kotlin.extensions.When
+import org.hamcrest.CoreMatchers.containsString
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.jupiter.api.*
+import spark.Spark
+import java.lang.Exception
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class RouterTest {
+    companion object {
+        const val PORT = 4444
+
+        val mockImageValidator = mockk<ImageValidator>()
+        val mockMemeManager = mockk<MemeManager>()
+        val mockTemplateReader = mockk<TemplateReader>()
+
+        @BeforeAll
+        @JvmStatic
+        fun startServer() {
+            Spark.port(PORT)
+
+            every { mockTemplateReader.read(any(), any()) } returns "index"
+
+            val router = Router(
+                    imageValidator = mockImageValidator,
+                    memeManager = mockMemeManager,
+                    reader = mockTemplateReader,
+                    domain = "domain")
+
+            router.start()
+        }
+
+        @AfterAll
+        @JvmStatic
+        fun stopServer() {
+            Spark.stop()
+        }
+    }
+
+    @BeforeEach
+    fun beforeEach() {
+        clearMocks(mockImageValidator, mockMemeManager, mockTemplateReader)
+    }
+    @Test
+    fun `should return 200 given get request with HTML content type`() {
+        Given {
+            port(PORT)
```

Comment:
You can assign a default port in `@BeforeAll` - this way you can avoid writing `port(PORT)` in each test. For example:
```kotlin
@BeforeAll
fun startServer() {
    ...

    RestAssured.port = PORT
}
```

Reply:
port is now set in beforeAll